test/pylib/schema_helper.py

```diff
@@ -0,0 +1,238 @@
+
+        if columns is not None:
+            assert len(columns) > pks, "Not enough value columns provided"
+            self.columns = columns
+        else:
+            assert ncolumns > pks, "Not enough value columns provided"
+            # Primary key pk, clustering columns c_xx, value columns v_xx
+            self.columns = [Column("pk", ctype=TextType)]
+            self.columns += [Column(f"c_{self.next_clustering_id():02}", ctype=TextType) for i in range(1, pks)]
+            self.columns += [Column(f"v_{self.next_value_id():02}") for i in range(1, ncolumns - pks + 1)]
+
+    @property
+    def all_col_names(self) -> str:
+        return ", ".join([c.name for c in self.columns])
+
+    async def create(self) -> asyncio.Future:
+        return await self.cql.run_async(f"CREATE TABLE {self.full_name} ("
+                                        + ", ".join(f"{c.cql}" for c in self.columns)
+                                        + ", primary key("
+                                        + ", ".join(c.name for c in self.columns[:self.pks]) + "))")
+
+    async def drop(self) -> asyncio.Future:
+        return await self.cql.run_async(f"DROP TABLE {self.full_name}")
+
+    def __str__(self):
+        return self.full_name
+
+
+class TestTables():
+    """Create tables for a test in a given keyspace"""
+    def __init__(self, test_name: str, cql: CassandraSession, keyspace: str, ntables: int, ncolumns: int):
+        self.test_name = test_name
+        self.cql = cql
+        self.keyspace = keyspace
+        self.tables: List[Table] = [Table(cql, keyspace, ncolumns) for _ in range(ntables)]
+        self.removed_tables: List[Table] = []
+
+    def new_table(self, ncolumns: int = None, columns: List[Column] = None, pks: int = 2, name: str = None) -> Table:
+        table = Table(self.cql, self.keyspace, ncolumns=ncolumns, columns=columns, pks=pks, name=name)
+        self.tables.append(table)
+        return table
+
+    def __getitem__(self, pos: int) -> Table:
+        return self.tables[pos]
+
+    def append(self, table: Table) -> None:
+        self.tables.append(table)
+
+    def extend(self, tables: List[Table]) -> None:
+        self.tables.extend(tables)
+
+    async def create_tables(self) -> None:
+        await asyncio.gather(*(t.create() for t in self.tables))
+
+    async def drop_table(self, table: Table) -> Table:
+        if type(table) is str:
+            table = next(t for t in self.tables if table in [t.name, t.full_name])
+        else:
+            assert type(table) is Table, f"Invalid table type {type(table)}"
+        await table.drop()
+        self.tables.remove(table)
+        self.removed_tables.append(table)
+        return table
+
+    async def drop_all_tables(self) -> None:
+        """Drop all current tables"""
+        await asyncio.gather(*(t.drop() for t in self.tables))
+        self.removed_tables.extend(self.tables)
+
+    async def verify_schema(self, table: Union[Table, str] = None) -> None:
```

Comment:
What's the purpose of this function?
If this is meant to *test* something, i.e., that the system_schema keyspace contains the right things, then this code should be in a test - not in a utility class - and this test should explain what it's testing and why.
If it's really a utility function, and not a test, I'm afraid I don't understand it's utility.

Reply:
I'm giving up. I'm personally not thrilled when complex highly-specific code that is designed for one test only is put in a library that is supposedly useful for many tests, but maybe I'm missing the bigger picture and you intend this to be used in additional tests as well?

In any case, I won't insist, but please do answer. Thanks.

Reply:
We need to test schema changes on top of topology changes when using Raft. The whole point of this file is to provide helpers to these upcoming tests so we don't need to manually write schema checks with CQL and tracking state of the table. Docstrings were added in v7.